src/config.rs

```diff
@@ -2162,6 +2190,11 @@ impl TiKvConfig {
             self.raft_store.raftdb_path = config::canonicalize_path(&self.raft_store.raftdb_path)?;
         }
 
+        let default_er_path = config::canonicalize_sub_path(&self.storage.data_dir, "raft-engine")?;
+        if self.raft_engine.config.dir.is_empty() {
+            self.raft_engine.config.dir = default_er_path;
+        }
```

Comment:
```suggestion
        let default_er_path = config::canonicalize_sub_path(&self.storage.data_dir, "raft-engine")?;
        if self.raft_engine.config.dir.is_empty() {
            self.raft_engine.config.dir = default_er_path;
        } else if self.raft_engine.config.dir != default_er_path {
            self.raft_engine.config.dir = config::canonicalize_path(&self.raft_engine.config.dir)?;
        }
```

Check out https://github.com/tikv/tikv/pull/6998#discussion_r390263181 https://github.com/tikv/tikv/pull/6998#discussion_r393707908